### PR TITLE
Feat(Timeline): add 'blacklist' / 'whitelist' for doc

### PR DESCRIPTION
**Pull request by stonebuzz**
## Checklist before requesting a review

*Please delete options that are not relevant.*

- [ ] I have read the CONTRIBUTING document.
- [ ] I have performed a self-review of my code.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] This change requires a documentation update.

## Description

Other proposal for https://github.com/glpi-project/glpi/pull/18026

Like `Edit` or `Delete permanently`, add new action `Blacklist for import` / `Whitelist for import` 

(For doc added from `followup` / `task` / `solution` or directly from `document` form)

When document is whitelisted

image

When document is blacklisted

image

@cedric-anne 

The `delete` action points to `ticket.form.php` but concerns the document.

Shouldn't we move the actions to the `document.form.php` ?

### CODE REVIEW

**Comment by cconard96** [comment]
I don't really understand the blacklisting of documents. Seems like if you upload a blacklisted document, it still gets attached to the item. Just looking at the code, it also seems handled backwards. In `Document::getDuplicateOf()`, being blacklisted is treated the same as if the file doesn't exist. So, wouldn't being blacklisted result in a duplicate Document being created?

**Comment by stonebuzz** [comment]
Here, the primary aim is to allow you to refuse to import the document directly from the timeline (without having to edit the document, change the value, save and then return to the ticket).

From memory, this option is only used when importing an e-mail (we use this option regularly, notably to avoid re-importing images in e-mail signatures).

File: front/ticket.form.php
Comment (cedric-anne) [inline front/ticket.form.php:254]: This should probably be moved in the `document.form.php` script.

File: templates/components/itilobject/timeline/form_document_item.html.twig
Comment (cedric-anne) [inline templates/components/itilobject/timeline/form_document_item.html.twig:93]: `blacklist`/`whitelist` terms should not be used, at least for new features.

```suggestion
                    {% set blacklisted_title    = entry_i['is_blacklisted'] ? _x('button', 'Remove from import exclusion list') : _x('button', 'Add to import exclusion list') %}
```

File: templates/components/itilobject/timeline/form_document_item.html.twig
Comment (cedric-anne) [inline templates/components/itilobject/timeline/form_document_item.html.twig:96]: It does not seems to be useful.

```suggestion
```

File: front/ticket.form.php
Comment (cedric-anne) [inline front/ticket.form.php:234]: This is not needed.

```suggestion
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: front/ticket.form.php
Comment (stonebuzz) [reply front/ticket.form.php]: :+1:  done !

File: templates/components/itilobject/timeline/form_document_item.html.twig
Comment (stonebuzz) [reply templates/components/itilobject/timeline/form_document_item.html.twig]: done !

File: front/ticket.form.php
Comment (stonebuzz) [reply front/ticket.form.php]: done !

**Review by AdrienClairembault — COMMENTED** [review]
Maybe we should just directly send an update instead of creating two new endpoints ?

**Comment by cedric-anne** [comment]
> Maybe we should just directly send an update instead of creating two new endpoints ?

Indeed, it should be possible to do it the existing update endpoint.

File: templates/components/itilobject/timeline/form_document_item.html.twig
Comment (AdrienClairembault) [inline templates/components/itilobject/timeline/form_document_item.html.twig:96]: The `reverse` filter is used to reverse strings and arrays, not toggle boolean: https://twig.symfony.com/doc/3.x/filters/reverse.html.

File: templates/components/itilobject/timeline/form_document_item.html.twig
Comment (cedric-anne) [inline templates/components/itilobject/timeline/form_document_item.html.twig:94]:
```suggestion
                     <form class="d-inline" method="post" action="{{ 'Document'|itemtype_form_path }}">
```

File: front/document.form.php
Comment (cedric-anne) [inline front/document.form.php:153]: This action should probably remain in the `ticket.form.php` script, as it is an action made on a specific ticket.

File: templates/components/itilobject/timeline/sub_documents.html.twig
Comment (cedric-anne) [inline templates/components/itilobject/timeline/sub_documents.html.twig:73]:
```suggestion
                     <form class="d-inline" method="post" action="{{ 'Document'|itemtype_form_path }}">
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: templates/components/itilobject/timeline/form_document_item.html.twig
Comment (stonebuzz) [reply templates/components/itilobject/timeline/form_document_item.html.twig]: fixed

File: front/document.form.php
Comment (stonebuzz) [reply front/document.form.php]: done

**Review by AdrienClairembault — APPROVED** [review]

**Comment by cedric-anne** [comment]
I propose to wait for the customer feedback before merging this PR.

**Comment by stonebuzz** [comment]
> I propose to wait for the customer feedback before merging this PR.

@flegastelois could you please test this feature, which you initially requested?

Best regards

**Pull request merged**
